runtime/src/consensus/state/staking.rs

```diff
@@ -0,0 +1,198 @@
+use std::collections::BTreeMap;
+
+use anyhow::anyhow;
+use io_context::Context;
+
+use crate::{
+    common::{
+        cbor,
+        key_format::{KeyFormat, KeyFormatAtom},
+        quantity::Quantity,
+    },
+    consensus::{
+        address::Address,
+        beacon::EpochTime,
+        staking::{Account, DebondingDelegation, Delegation},
+        state::StateError,
+    },
+    key_format,
+    storage::mkvs::ImmutableMKVS,
+};
+
+/// Consensus staking state wrapper.
+pub struct ImmutableState<'a, T: ImmutableMKVS> {
+    pub mkvs: &'a T,
+}
```

Comment:
Maybe add a constructor so you don't need to initialize the struct each time?

Reply:
:+1: